persist/fs/msgpack/decoder.go

```diff
@@ -215,17 +221,25 @@ func (dec *Decoder) DecodeLogMetadata() (schema.LogMetadata, error) {
 }
 
 func (dec *Decoder) decodeIndexInfo() schema.IndexInfo {
-	numFieldsToSkip, ok := dec.checkNumFieldsFor(indexInfoType)
+	numFieldsToSkip, actual, ok := dec.checkNumFieldsFor(indexInfoType)
 	if !ok {
 		return emptyIndexInfo
 	}
+
 	var indexInfo schema.IndexInfo
-	indexInfo.Start = dec.decodeVarint()
+	indexInfo.BlockStart = dec.decodeVarint()
 	indexInfo.BlockSize = dec.decodeVarint()
 	indexInfo.Entries = dec.decodeVarint()
 	indexInfo.MajorVersion = dec.decodeVarint()
 	indexInfo.Summaries = dec.decodeIndexSummariesInfo()
 	indexInfo.BloomFilter = dec.decodeIndexBloomFilterInfo()
+
+	if actual < 8 {
```

Comment:
nit: could you use the `currNum...` value here?

Reply:
It works in this case, but I don't think its a good idea. You can imagine that if we added another field then we would want a check thats like 

```golang
if actual < 8 {
  ...
}

if actual < 9 {
  ...
}
```

etc etc

Reply:
Yeah, explicitly calling out `< X` sets up a good pattern for the future.